.github/scripts/test_check_pr_quality.py

```diff
@@ -0,0 +1,377 @@
+        title = "Fixed bug in authentication"
+        body = "This PR fixes the authentication issue"
+        issues = check_pr_quality(title, body)
+        self.assertEqual(len(issues), 1)
+        self.assertEqual(issues[0].type, IssueType.MISSING_TICKET)
+        self.assertEqual(issues[0].severity, Severity.HIGH)
+        self.assertEqual(len(issues[0].matches), 0)
+
+    def test_tutorial_pattern_in_title(self):
+        title = "My first contribution to Django"
+        body = "This is a real fix"
+        issues = check_pr_quality(title, body)
+
+        # Should have both missing ticket and tutorial issues
+        self.assertEqual(len(issues), 2)
+        issue_types = [issue.type for issue in issues]
+        self.assertIn(IssueType.MISSING_TICKET, issue_types)
+        self.assertIn(IssueType.TUTORIAL, issue_types)
+
+        # Check tutorial issue details
+        tutorial_issue = next(
+            issue for issue in issues if issue.type == IssueType.TUTORIAL
+        )
+        self.assertEqual(len(tutorial_issue.matches), 1)
+        self.assertEqual(tutorial_issue.matches[0].pattern, "first contribution")
+        self.assertEqual(tutorial_issue.matches[0].location, Location.TITLE)
+
+    def test_high_confidence_pattern_in_body(self):
+        title = "Fix authentication bug #12345"
+        body = "This is my first pr to Django. I followed some guide."
+        issues = check_pr_quality(title, body)
+
+        self.assertEqual(len(issues), 1)
+        self.assertEqual(issues[0].type, IssueType.TUTORIAL)
+
+        # Should find "first pr" (high confidence, checked in body)
+        self.assertEqual(len(issues[0].matches), 1)
+        self.assertEqual(issues[0].matches[0].pattern, "first pr")
+        self.assertEqual(issues[0].matches[0].location, Location.BODY)
+
+    def test_low_confidence_pattern_not_in_body(self):
+        title = "Add test for authentication"
+        body = "This adds a comprehensive test suite for the authentication module"
+        issues = check_pr_quality(title, body)
+
+        self.assertEqual(len(issues), 2)  # missing_ticket + tutorial
+        tutorial_issue = next(
+            issue for issue in issues if issue.type == IssueType.TUTORIAL
+        )
+
+        # Should only find "test" in title, not in body (low confidence)
+        self.assertEqual(len(tutorial_issue.matches), 1)
+        self.assertEqual(tutorial_issue.matches[0].location, Location.TITLE)
+        self.assertEqual(tutorial_issue.matches[0].pattern, r"\btest\b")
+
+    def test_multiple_patterns_in_title(self):
+        title = "My first contribution - learning Django test"
+        body = "This is a real change"
+        issues = check_pr_quality(title, body)
+
+        tutorial_issue = next(
+            issue for issue in issues if issue.type == IssueType.TUTORIAL
+        )
+        self.assertEqual(
+            len(tutorial_issue.matches), 3
+        )  # first contribution, learning, test
```

Comment:
The comment here isn't useful, especially since we `assert` exactly these things below.

```suggestion
        self.assertEqual(len(tutorial_issue.matches), 3)
```